odoo/tools/translate.py

```diff
@@ -889,6 +892,65 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_formula_terms(formula):
+    """Extract strings in a spreadsheet formula which are arguments to '_t' functions
+
+        >>> extract_formula_terms('=_t("Hello") + _t("Raoul")')
+        ["Hello", "Raoul"]
+    """
+    tokens = generate_tokens(io.StringIO(formula).readline)
+    tokens = (token for token in tokens if token.type not in {NEWLINE, INDENT, DEDENT})
+    for t1 in tokens:
+        if not t1.string == '_t':
+            continue
+        t2 = next(tokens, None)
+        if t2 and t2.string == '(':
+            t3 = next(tokens, None)
+            t4 = next(tokens, None)
+            if t4 and t4.string == ')' and t3 and t3.type == STRING:
+                yield t3.string[1:][:-1] # strip leading and trailing quotes
```

Comment:
Hi @LucasLefevre 
I have a small question
Is `extract.extract('javascript', io.BytesIO(formula.encode()), keywords={'_t': None, '_lt': None}, options={}))` considered when you develope the feature?
It may be slower but you can concatenate all `formula` into one string and extract them at once.
I think it will be safer than the current implementation, and you can write the formula like js code

Reply:
Hello @HydrionBurst ,
No it was not considered. I'm not very familiar with babel.
Note they are *not* javascript expressions. They are spreadsheet expressions.
But they use a limited number of expressions. I think they are valid tokens both for python and javascript